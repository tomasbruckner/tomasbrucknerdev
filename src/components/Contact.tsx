import React, { FC } from 'react';
import { makeStyles, Typography } from '@material-ui/core';
import MediumIcon from './icons/MediumIcon';
import GithubIcon from './icons/GithubIcon';
import StackOverflowIcon from './icons/StackOverflowIcon';
import LinkedinIcon from './icons/LinkedinIcon';
import ContactEmailIcon from './icons/ContactEmailIcon';
import { useCommonStyles } from '../styles/commonStyles';
import { t } from '../utils/i18n';

const useStyles = makeStyles((theme) => ({
  contact: {
    marginTop: theme.spacing(2),
    display: 'flex',
    alignItems: 'center',
  },
  text: {
    marginTop: theme.spacing(2),
  },
}));

const Contact: FC = () => {
  const classes = useStyles();
  const commonClasses = useCommonStyles();

  return (
    <>
      <Typography variant="h4" id="contact" gutterBottom className={classes.text}>
        {t('navigation.contact')}
      </Typography>
      <Typography gutterBottom className={classes.contact}>
        <GithubIcon />
        <a className={commonClasses.linkStyle} href="https://github.com/tomasbruckner">
          github.com/tomasbruckner
        </a>
      </Typography>
      <Typography gutterBottom className={classes.contact}>
        <StackOverflowIcon />
        <a
          className={commonClasses.linkStyle}
          href="https://stackoverflow.com/users/3323141/tomas-bruckner"
        >
          stackoverflow.com/users/3323141
        </a>
      </Typography>
      <Typography gutterBottom className={classes.contact}>
        <MediumIcon />
        <a className={commonClasses.linkStyle} href="https://medium.com/@tomasbruckner">
          medium.com/@tomasbruckner
        </a>
      </Typography>
      <Typography gutterBottom className={classes.contact}>
        <LinkedinIcon />
        <a className={commonClasses.linkStyle} href="https://www.linkedin.com/in/tomasbruckner">
          linkedin.com/in/tomasbruckner
        </a>
      </Typography>
      <Typography gutterBottom className={classes.contact}>
        <ContactEmailIcon />
        <a className={commonClasses.linkStyle} href="mailto:tomasbrucknermail@gmail.com">
          tomasbrucknermail@gmail.com
        </a>
      </Typography>
    </>
  );
};

export default Contact;
