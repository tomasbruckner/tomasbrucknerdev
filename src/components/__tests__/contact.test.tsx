import React from 'react';
import Contact from '../Contact';
import renderer from 'react-test-renderer';

it('Contact renders correctly', () => {
  const tree = renderer.create(<Contact />).toJSON();
  expect(tree).toMatchSnapshot();
});
