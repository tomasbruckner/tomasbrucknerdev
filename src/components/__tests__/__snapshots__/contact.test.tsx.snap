// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact renders correctly 1`] = `
Array [
  <h4
    className="MuiTypography-root makeStyles-text-2 MuiTypography-h4 MuiTypography-gutterBottom"
    id="contact"
  >
    Kontakt
  </h4>,
  <p
    className="MuiTypography-root makeStyles-contact-1 MuiTypography-body1 MuiTypography-gutterBottom"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12.007 0C6.12 0 1.1 4.27.157 10.08c-.944 5.813 2.468 11.45 8.054 13.312.19.064.397.033.555-.084.16-.117.25-.304.244-.5v-2.042c-3.33.735-4.037-1.56-4.037-1.56-.22-.726-.694-1.35-1.334-1.756-1.096-.75.074-.735.074-.735.773.103 1.454.557 1.846 1.23.694 1.21 2.23 1.638 3.45.96.056-.61.327-1.178.766-1.605-2.67-.3-5.462-1.335-5.462-6.002-.02-1.193.42-2.35 1.23-3.226-.327-1.015-.27-2.116.166-3.09 0 0 1.006-.33 3.3 1.23 1.966-.538 4.04-.538 6.003 0 2.295-1.5 3.3-1.23 3.3-1.23.445 1.006.49 2.144.12 3.18.81.877 1.25 2.033 1.23 3.226 0 4.607-2.805 5.627-5.476 5.927.578.583.88 1.386.825 2.206v3.29c-.005.2.092.393.26.507.164.115.377.14.565.063 5.568-1.88 8.956-7.514 8.007-13.313C22.892 4.267 17.884.007 12.008 0z"
      />
    </svg>
    <a
      className="makeStyles-linkStyle-4"
      href="https://github.com/tomasbruckner"
    >
      github.com/tomasbruckner
    </a>
  </p>,
  <p
    className="MuiTypography-root makeStyles-contact-1 MuiTypography-body1 MuiTypography-gutterBottom"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 120 120"
    >
      <path
        d="M84.4 93.8V70.6h7.7v30.9H22.6V70.6h7.7v23.2z"
        style={
          Object {
            "fill": "#bcbbbb",
          }
        }
      />
      <path
        d="M38.8 68.4l37.8 7.9 1.6-7.6-37.8-7.9-1.6 7.6zm5-18l35 16.3 3.2-7-35-16.4-3.2 7.1zm9.7-17.2l29.7 24.7 4.9-5.9-29.7-24.7-4.9 5.9zm19.2-18.3l-6.2 4.6 23 31 6.2-4.6-23-31zM38 86h38.6v-7.7H38V86z"
        style={
          Object {
            "fill": "#f48023",
          }
        }
      />
    </svg>
    <a
      className="makeStyles-linkStyle-4"
      href="https://stackoverflow.com/users/3323141/tomas-bruckner"
    >
      stackoverflow.com/users/3323141
    </a>
  </p>,
  <p
    className="MuiTypography-root makeStyles-contact-1 MuiTypography-body1 MuiTypography-gutterBottom"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      clipRule="evenodd"
      fillRule="evenodd"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 0c-6.626 0-12 5.372-12 12 0 6.627 5.374 12 12 12 6.627 0 12-5.373 12-12 0-6.628-5.373-12-12-12zm6.974 17.557v-.235l-1.092-1.072c-.096-.073-.144-.194-.124-.313v-7.874c-.02-.119.028-.24.124-.313l1.118-1.072v-.235h-3.869l-2.758 6.88-3.138-6.88h-4.059v.235l1.308 1.575c.128.115.194.285.176.457v6.188c.038.223-.032.451-.189.614l-1.471 1.784v.235h4.17v-.235l-1.471-1.784c-.158-.163-.233-.389-.202-.614v-5.352l3.66 7.985h.425l3.143-7.985v6.365c0 .17 0 .202-.111.313l-1.13 1.098v.235h5.49z"
      />
    </svg>
    <a
      className="makeStyles-linkStyle-4"
      href="https://medium.com/@tomasbruckner"
    >
      medium.com/@tomasbruckner
    </a>
  </p>,
  <p
    className="MuiTypography-root makeStyles-contact-1 MuiTypography-body1 MuiTypography-gutterBottom"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 0h-14c-2.761 0-5 2.239-5 5v14c0 2.761 2.239 5 5 5h14c2.762 0 5-2.239 5-5v-14c0-2.761-2.238-5-5-5zm-11 19h-3v-11h3v11zm-1.5-12.268c-.966 0-1.75-.79-1.75-1.764s.784-1.764 1.75-1.764 1.75.79 1.75 1.764-.783 1.764-1.75 1.764zm13.5 12.268h-3v-5.604c0-3.368-4-3.113-4 0v5.604h-3v-11h3v1.765c1.396-2.586 7-2.777 7 2.476v6.759z"
      />
    </svg>
    <a
      className="makeStyles-linkStyle-4"
      href="https://www.linkedin.com/in/tomasbruckner"
    >
      linkedin.com/in/tomasbruckner
    </a>
  </p>,
  <p
    className="MuiTypography-root makeStyles-contact-1 MuiTypography-body1 MuiTypography-gutterBottom"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M21 8V7l-3 2-3-2v1l3 2 3-2zm1-5H2C.9 3 0 3.9 0 5v14c0 1.1.9 2 2 2h20c1.1 0 1.99-.9 1.99-2L24 5c0-1.1-.9-2-2-2zM8 6c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm6 12H2v-1c0-2 4-3.1 6-3.1s6 1.1 6 3.1v1zm8-6h-8V6h8v6z
"
      />
    </svg>
    <a
      className="makeStyles-linkStyle-4"
      href="mailto:tomasbrucknermail@gmail.com"
    >
      tomasbrucknermail@gmail.com
    </a>
  </p>,
]
`;
